### DISCUSSION TITLE
Feature: drop down menu like in Microsoft Excel

### DISCUSSION THREAD

**Discussion by poetaman** [andmarti1424/sc-im#701] [Ideas]
Recently a friend showed me a feature of Microsoft Excel that is pretty nifty, and saves time for a class of sheets. One can define drop down menu for cell, and click on the cell to show the possible valid values. Perhaps this can be implemented using json. For instance, user will be able to specify a directory to lookup, and that directory will have jsons that just have ``"key":"value"`` pairs. If a user clicks a certain key over a cell, it becomes a drop-down (i.e form) cell, and will show the name of jsons (from the user specified directory) that user can select from. After a json has been assigned to a cell, if a user hits a key, they can select a key, and it's a value will be plugged in to the cell. Copying the cell to other cells copies the json/value and user can change the value from the drop-down again. Thanks!

An extension could be to allow an external program to help with selection of key/value for a cell, like via fzf, to leverage the search capability of fzf (to search key/values in json).

**Comment by andmarti1424** [comment]
hello. this is a really specific case. dont know if a lot of people would use something like this.
i know of fzf. cool tool btw.
move this to Discussion -> ideas / FR so others can give their opinions.

**Reply by airaqi** [reply]
actually, having a fuzzy find function would be a very useful, either just finding and replacing data, or as formula function.
